### PR TITLE
Adding new bootstrap options to knife-windows from Chef 12

### PR DESCRIPTION
**Pull request by Vasu1105**
- Added --bootstrap-vault-json, --bootstrap-vault-item and --bootstrap-vault-file options

### CODE REVIEW

**Comment by Vasu1105** [comment]
@btm please review .

File: spec/unit/knife/bootstrap_options_spec.rb [hunk 1]
```diff
@@ -84,6 +84,9 @@ def compare_property(sym)
       :winrm_password => :ssh_password,
       :winrm_port => :ssh_port,
       :winrm_ssl_verify_mode => :host_key_verify,
+      :bootstrap_vault_file => :bootstrap_vault_file,
+      :bootstrap_vault_item => :bootstrap_vault_item,
+      :bootstrap_vault_json => :bootstrap_vault_json,
```
Comment (btm) [inline spec/unit/knife/bootstrap_options_spec.rb]: You shouldn't need these mappings, these are only for when the options are different.

File: spec/unit/knife/bootstrap_options_spec.rb [hunk 2]
```diff
@@ -126,6 +126,9 @@ def compare_property(sym)
       :msi_url => :bootstrap_url,
       :encrypted_data_bag_secret => :secret,
       :encrypted_data_bag_secret_file => :secret_file,
+      :bootstrap_vault_file => :bootstrap_vault_file,
+      :bootstrap_vault_item => :bootstrap_vault_item,
+      :bootstrap_vault_json => :bootstrap_vault_json,
```
Comment (btm) [inline spec/unit/knife/bootstrap_options_spec.rb]: Same as above, this mapping is equivalent and shouldn't be necessary.

File: lib/chef/knife/bootstrap/windows-chef-client-msi.erb [hunk 3]
```diff
@@ -176,7 +176,6 @@ goto install
 @echo Installing downloaded client package...
 
 <%= install_chef %>
-
```
Comment (btm) [inline lib/chef/knife/bootstrap/windows-chef-client-msi.erb]: I think this newline is nice for clarity, perhaps an accidental change?

**Comment by btm** [comment]
rebased to drop the extra commit + revert, committed as  43d0489911cc49196c208d56add433bdce51287f with my suggested edits in 4aa4244179fd67ff2b3f1540fc866a50579f7e85. Thanks @Vasu1105.

**Pull request closed without merging**